psutil/_psutil_posix.c

```diff
@@ -429,6 +430,186 @@ psutil_net_if_mtu(PyObject *self, PyObject *args) {
     return PyErr_SetFromErrno(PyExc_OSError);
 }
 
+static bool
+check_and_append_iff_flag(PyObject *py_retlist, short int flags, short int flag_to_check, const char * flag_name)
+{
+    PyObject *py_str = NULL;
+
+    if (flags & flag_to_check) {
+        py_str = PyUnicode_DecodeFSDefault(flag_name);
+        if (! py_str)
+            return false;
+        if (PyList_Append(py_retlist, py_str)) {
+            Py_DECREF(py_str);
+            return false;
+        }
+        Py_CLEAR(py_str);
+    }
+
+    return true;
+}
```

Comment:
I don't like doing the flag comparison / check in here (`if flags & flag_to_check ...`). That should happen in the main function. The utility function (this one) should only add the item to the dict.

Also, instead of return `true` or `false` I would just return `1` or `0`_, which is consistent with the rest of the code base and avoids including `<stdbool.h>` (I have some memories of this header file causing problems on some platform).

Reply:
That's fair.  I've switched this helper function to return an int, and I'm now doing the check in the main function.  All in d83c959a981c68ff7ff3d60a155b4f35640e9fa9.